### PR TITLE
Added export command. Refactor code for parse command. Minor fixes and improvements

### PR DESCRIPTION
**Pull request by rawmind0**
This PR contains:
- Refactor code adding debug option and working with distinct commands

```
# migration-tools -h
NAME:
   Rancher 1.6 to Rancher 2.0 migration-helper - Please check the options using --help flag

USAGE:
   migration-tools [global options] command [command options] [arguments...]

VERSION:
   git

AUTHOR:
   Rancher Labs, Inc.

COMMANDS:
     export   Export compose files for every stack running on cattle environment on a Rancher v1.6 system
     parse    Parse docker-compose and rancher-compose files to get k8s manifests
     help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --debug        debug logging
   --log value    path to log to
   --help, -h     show help
   --version, -v  print the version
```

- Current code working as `migration-tools parse` command.

```
# migration-tools parse -h
NAME:
   migration-tools parse - Parse docker-compose and rancher-compose files to get k8s manifests

USAGE:
   migration-tools parse [command options] [arguments...]

OPTIONS:
   --docker-file value   Docker compose file to parse to get k8s manifest (default: "docker-compose.yml")
   --output-file value   Output file where to write checks and advices for conversion (default: "output.txt")
   --rancher-file value  Rancher compose file to parse to get k8s manifest (default: "rancher-compose.yml")
```

- Moved and ordered code to `cmd` folder.
- PR https://github.com/rancher/migration-tools/pull/7 already included in the code
- Added new `migration-tools export` command to connect to running Rancher v1.6 system and export compose files for every stack.

```
# migration-tools export -h
NAME:
   migration-tools export - Export compose files for every stack running on cattle environment on a Rancher v1.6 system

USAGE:
   migration-tools export [command options] [arguments...]

OPTIONS:
   --url value         Rancher API endpoint URL [$RANCHER_URL]
   --access-key value  Rancher API access key. Using admin API key will export stacks on all cattle environments [$RANCHER_ACCESS_KEY]
   --secret-key value  Rancher API secret key [$RANCHER_SECRET_KEY]
   --export-dir value  Base directory under which compose files will be exported under sub-directories created for every env/stack (default: "export")
   --all, -a           Export all stacks. Using this flag stacks with inactive, stopped and removing state, will also be exported
   --system, -s        Export system and infrastructure stacks
```

- Updated `README.md`. 
- Updated vendor files with go-rancher library.

Related to issue https://github.com/rancher/rancher/issues/15454

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,35 +2,91 @@ migration-tools
...
-
-**Options**:
-
--   `--docker-file value`   An absolute path to an alternate Docker compose file (default: "docker-compose.yml")
--   `--rancher-file value`  An absolute path to an alternate Rancher compose file (default: "rancher-compose.yml")
--   `--help, -h`           show help
--   `--version, -v`        print the version
+This tool will:
+- Export docker compose config files [docker-compose.yml and rancher-compose.yml] for every stack running on cattle environments on existing Rancher v1.6 system.
+- Convert docker compose config files and output a list of constructs present in the config file that cannot be supported onto Rancher 2.0 without special handling or that cannot be converted to Kubernetes YAML using [Kompose tool](https://github.com/kubernetes/kompose) tool.
```
Comment (prachidamle) [inline README.md]: We do not really convert the files, so maybe we can say parse the docker compose config files?

File: README.md [hunk 2]
```diff
@@ -2,35 +2,91 @@ migration-tools
...
+   migration-tools [global options] command [command options] [arguments...]
+
+VERSION:
+   git
+
+AUTHOR:
+   Rancher Labs, Inc.
+
+COMMANDS:
+     convert  Convert docker-compose and rancher-compose into k8s manifests
```
Comment (prachidamle) [inline README.md]: can we say parse?

File: README.md [hunk 3]
```diff
@@ -2,35 +2,91 @@ migration-tools
...
+# migration-tools convert -h
+NAME:
+   migration-tools convert - Convert docker-compose and rancher-compose into k8s manifests
+
+USAGE:
+   migration-tools convert [command options] [arguments...]
+
+OPTIONS:
+   --docker-file value   Docker compose file to convert to k8s manifest (default: "docker-compose.yml")
+   --output-file value   Output file where to write checks and advices for convertion (default: "output.txt")
```
Comment (prachidamle) [inline README.md]: nit: replace convertion by conversion

File: README.md [hunk 4]
```diff
@@ -2,35 +2,91 @@ migration-tools
...
+   migration-tools export - Export compose files for every stack running on cattle environment on a Rancher v1.6 system
+
+USAGE:
+   migration-tools export [command options] [arguments...]
+
+OPTIONS:
+   --url value         Rancher API endpoint URL [$RANCHER_URL]
+   --access-key value  Rancher API access key. Using admin API key will export stacks on all cattle environments [$RANCHER_ACCESS_KEY]
+   --secret-key value  Rancher API secret key [$RANCHER_SECRET_KEY]
+   --export-dir value  Directory where to export compose files for every stack (default: "export")
```
Comment (prachidamle) [inline README.md]: Maybe say that this is the "Base" directory under which compose files will be exported under sub-directories created for every env/stack

File: cmd/convert.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+func ConvertCommand() cli.Command {
+	return cli.Command{
+		Name:   "convert",
```
Comment (prachidamle) [inline cmd/convert.go]: Again, we don't want to claim that we are converting docker-compose files to k8s YAML. We are parsing to output what is lacking and as an Addon also creating first-level K8s specs that Kompose would generate

Comment (prachidamle) [reply cmd/convert.go]: So can we rename all Convert usage to Parse or ParseCompose  or ReadCompose

**Comment by prachidamle** [comment]
The export command code looks good @rawmind0, just have some comments on name of the other command

File: README.md (hunk 1 above)
Comment (rawmind0) [reply README.md]: Ok

File: README.md (hunk 2 above)
Comment (rawmind0) [reply README.md]: Ok

File: README.md (hunk 3 above)
Comment (rawmind0) [reply README.md]: Ok

File: README.md (hunk 4 above)
Comment (rawmind0) [reply README.md]: Ok

File: cmd/convert.go (hunk 5 above)
Comment (rawmind0) [reply cmd/convert.go]: ok.. renamed command to parse and file to `cmd/parse.go`

**Comment by rawmind0** [comment]
@prachidamle, implemented your requests, renamed `convert` command to `parse` and corrected typos on `README.md`.

**Comment by prachidamle** [comment]
LGTM

**Comment by prachidamle** [comment]
@alena1108 can you merge this in

**Pull request merged**